packages/api/amplify_api/test/util.dart

```diff
@@ -150,21 +150,39 @@ class MockWebSocketConnection extends WebSocketConnection {
           );
           // start, respond with start_ack and mock data
         } else if (messageFromEvent.messageType == MessageType.start) {
-          mockResponseMessages
-            ..add(
-              WebSocketMessage(
-                messageType: MessageType.startAck,
-                id: messageFromEvent.id,
-              ),
-            )
-            ..add(
-              WebSocketMessage(
-                messageType: MessageType.data,
-                id: messageFromEvent.id,
-                payload:
-                    const SubscriptionDataPayload(mockSubscriptionData, null),
-              ),
+          if (!event.contains('Invalid')) {
```

Comment:
This logic is going to need refactoring when reconnection gets merged. This whole function block will no longer exist. Mock messages now get put directly onto the channel stream. An example of how I replaced response mocking is [here](https://github.com/Equartey/amplify-flutter/blob/b1f162b1ea9c1cd3e73d08682bdb83f0ddb01b31/packages/api/amplify_api/test/util.dart#L164)

Reply:
Couldn't see a good way to mock this w new mock impl so just did it with integration test.